u4-databases/features/uni-controller.js

```diff
@@ -0,0 +1,50 @@
+const { db } = require('../_services/firebase-initialized')
+
+const iterationList = async (req, res) => {
+  const query = db.collection('iterations').get()
+  const iteration = (await query).docs.map(doc => doc.data())
+  res.render('uni-iteration-list', { iteration })
+}
+
+const iterationDetails = async (req, res) => {
+  const code = req.params.code
+  const query = db.collection('iterations').doc(code).get()
+  const iteration = (await query).data()
+  res.render('uni-iteration-details', { iteration })
+}
+
+const studentDetails = async (req, res) => {
+  const code = req.params.code
+  const query = db.collection('students').doc(code).get()
+  const students = (await query).data()
+  res.render('uni-student-details', { students })
+}
+
+const courseList = async (req, res) => {
+  const query = db.collection('iterations')
+    .where('year', '==', 2021)
+    .where('semester', '==', 1)
+    .get()
+  const iteration = (await query).docs.map(doc => doc.data())
+  res.render('uni-course-list', { iteration })
+}
+
+const courseDetails = async (req, res) => {
+  const code = req.params.code
```

Comment:
เพื่อหลีกเลี่ยงความสับสน คุณสามารถตั้งชื่อ `courseCode` นี้
To avoid confusion, you could name this `courseCode`.